dipy/core/gradients.py

```diff
@@ -67,7 +69,61 @@ def __init__(self, gradients, big_delta=None, small_delta=None,
         self.big_delta = big_delta
         self.small_delta = small_delta
         self.b0_threshold = b0_threshold
+        b_tensors = np.zeros((len(self.bvals), 3, 3))
+        if isinstance(btens, str):
+            if btens == 'LTE':
+                b_tensor = np.array([[1, 0, 0],
+                                     [0, 0, 0],
+                                     [0, 0, 0]])
+            elif btens == 'PTE':
+                b_tensor = np.array([[0, 0, 0],
+                                     [0, 1, 0],
+                                     [0, 0, 1]]) / 2
+            elif btens == 'STE':
+                b_tensor = np.array([[1, 0, 0],
+                                     [0, 1, 0],
+                                     [0, 0, 1]]) / 3
+            else:
+                raise ValueError("Invalid value for btens")
+            for i, (bvec, bval) in enumerate(zip(self.bvecs, self.bvals)):
+                if btens == 'STE':
+                    b_tensors[i] = b_tensor * bval
+                else:
+                    R = vec2vec_rotmat(np.array([1, 0, 0]), bvec)
+                    b_tensors[i] = (np.matmul(np.matmul(R, b_tensor), R.T)
+                                    * bval)
+        elif (isinstance(btens, np.ndarray) and (btens.shape == 
+                (gradients.shape[0],) or (btens.shape == 
+                (gradients.shape[0],1)) or (btens.shape == (1,
+                gradients.shape[0])))):
+            if (btens.shape == (1,gradients.shape[0],1)):
+                btens = btens.reshape((gradients.shape[0],1))
+            for i, (bvec, bval) in enumerate(zip(self.bvecs, self.bvals)):
+                R = vec2vec_rotmat(np.array([1, 0, 0]), bvec)
+                if btens[i] == 'LTE':
+                    b_tensor = np.array([[1, 0, 0],
+                                         [0, 0, 0],
+                                         [0, 0, 0]])
+                    b_tensors[i] = (np.matmul(np.matmul(R, b_tensor), R.T)
+                                    * bval)
+                elif btens[i] == 'PTE':
+                    b_tensor = np.array([[0, 0, 0],
+                                         [0, 1, 0],
+                                         [0, 0, 1]]) / 2
+                    b_tensors[i] = (np.matmul(np.matmul(R, b_tensor), R.T)
+                                    * bval)
+                elif btens[i] == 'STE':
+                    b_tensor = np.array([[1, 0, 0],
+                                         [0, 1, 0],
+                                         [0, 0, 1]]) / 3
+                    b_tensors[i] = b_tensor * bval
+                else:
+                    raise ValueError("Invalid value in btens array")     
+        else:
+                raise ValueError("Invalid value for btens")
```

Comment:
Again: probably want better error handling here. Tell the users what they gave you (they might not know, because there's another layer between them and this code!) and what they should give you.